### ISSUE TITLE
Adjust Threadpool

### ISSUE THREAD

**Issue by lnfn** [adjust/android_sdk#622]
Adjust consumes a lot of resources. In particular, it creates many Threads
Can you make improvements so that you can use the Adjust SDK to set the Threadpool to save resources.

**Comment by nonelse** [comment]
Hi, could you be more specific next time? This issue does not indicate what is the problem, we have looked into alternative thread management possibilities, but figured the current system is adequate. Feel free to open a new issue with more detail, thanks.

**Comment by TheKeeperOfPie** [comment]
The problem here is `SingleThreadCachedScheduler` and related. Each component instantiates one of these: https://github.com/adjust/android_sdk/blob/b31ee274a2d189c4ff705c2ee6c47ad09ca5eb62/Adjust/sdk-core/src/main/java/com/adjust/sdk/AttributionHandler.java#L62

Which creates a bunch of threads that remain cached forever while the SDK is alive. Even if they don't actively do any work, a thread incurs a memory cost, and it's unnecessarily spawned when the app can provide its own thread pool (like `Dispatchers.Default/IO`) to perform this work. That would allow sharing with other app functions.

Basically the ask here is to allow specifying an executor/dispatcher pool for these threads. Ideally this could be a physical `CoroutineDispatcher` input, but asking for a Kotlin migration is probably out of the question. So something like `RxJavaPlugins` which can static swap the pool used would be enough, and we can call that before initializing the SDK.
